### PR TITLE
Packaging updates

### PR DESCRIPTION
**Pull request by bhomnick**
- Flake8 everything
- Clean up method comments
- Restore test suite, fix failing tests
- Add CI script (will need to enable the CircleCI project for the FactomProject org)
- Drop Python 2.7 support, add Python 3.8 support
- Slight doc updates
- Bump version

### CODE REVIEW

**Comment by PaulBernier** [comment]
Thanks a lot for the PR. Will try to get a Pythonist to review it.

**Review by sanchopansa — APPROVED** [review]
Thanks @bhomnick for those changes! LGTM.

I would like to see the CircleCI integration working first before we merge the PR, but unfortunately I don't have the permissions to set this up (afaik you need to be an admin of the FactomProject to do this?).

Maybe @stackdump or @PaulSnow could help with this?

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+version: 2.1
+jobs:
+  build:
+    machine:
+      image: ubuntu-1604:201903-01
```
Comment (sanchopansa) [inline .circleci/config.yml:5]: Minor: any reason not to use a more recent version of Ubuntu as the image?

**Comment by bhomnick** [comment]
This seems to be the recommended version, according to the docs:
https://circleci.com/docs/2.0/executor-types/#using-machine

**Pull request merged**